GameServerLib/Logic/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -60,29 +62,97 @@ public override void OnRemoved()
             _game.ObjectManager.RemoveVisionUnit(this);
         }
 
-        public Stats GetStats()
-        {
-            return Stats;
-        }
-
         public override void update(float diff)
         {
             base.update(diff);
 
+            _healthUpdateTimer += diff;
             _statUpdateTimer += diff;
-            while (_statUpdateTimer >= 500)
-            { // update Stats (hpregen, manaregen) every 0.5 seconds
-                Stats.update(_statUpdateTimer);
-                _statUpdateTimer -= 500;
+            if (_healthUpdateTimer >= 500)
+            {
+                if (Stats.TotalHealthRegen > 0 && Stats.CurrentHealth < Stats.TotalHealth && Stats.CurrentHealth > 0)
+                {
+                    Stats.CurrentHealth = Math.Min(Stats.TotalHealth,
+                        Stats.CurrentHealth + Stats.TotalHealthRegen * diff * 0.001f);
+                }
+
+                if (Stats.TotalParRegen > 0 && Stats.CurrentPar < Stats.TotalPar && Stats.CurrentPar > 0)
+                {
+                    Stats.CurrentPar = Math.Min(Stats.TotalPar,
+                        Stats.CurrentPar + Stats.TotalParRegen * diff * 0.001f);
+                }
+
+                UpdateReplication();
+                _healthUpdateTimer = 0;
+            }
+
+            if (_statUpdateTimer >= 100)
+            {
+                UpdateReplication();
+                _statUpdateTimer = 0;
             }
         }
 
-        public override float getMoveSpeed()
+        public virtual void UpdateReplication()
+        {
+
+        }
+
+        public virtual bool GetTargetableToTeam(TeamId team)
         {
-            return Stats.MoveSpeed.Total;
+            if (Stats.IsTargetableToTeam.HasFlag(IsTargetableToTeamFlags.TargetableToAll))
```

Comment:
I would leave ``TargetableToAll`` set to true allways at least utill we fully understand what it does.
If my guess is correct its used by GP barrels(set to false) to make a unit non-targetable for everyone else but him.
```
        public virtual bool GetTargetableToTeam(TeamId team)
        {
            if (!Stats.IsTargetable)
            {
                return false;
            }
            if(team == Team)
            {
                return (Stats.IsTargetableToTeam & IsTargetableToTeamFlags.NonTargetableAlly) == 0;
            }
            else
            {
                return(Stats.IsTargetableToTeam & IsTargetableToTeamFlags.NonTargetableAlly) == 0;
            }
        }
```